pkg/cache/metrics.go

```diff
@@ -72,37 +77,69 @@ func init() {
 			}, []string{"cache_name"}),
 		}
 	})
+
+	// Initialize the singleton EvictionMetrics.
+	// Set up Prometheus counters for cache evictions.
+	evictionCollectorOnce.Do(func() {
+		evictionCollector = &EvictionMetrics{
+			evictions: promauto.NewCounterVec(prometheus.CounterOpts{
+				Namespace: common.MetricsNamespace,
+				Subsystem: common.MetricsSubsystem,
+				Name:      "evictions_total",
+				Help:      "Total number of cache evictions.",
+			}, []string{"cache_name"}),
+		}
+	})
 }
 
 var (
-	collectorOnce sync.Once // Ensures that the collector is initialized only once.
-	collector     *MetricsCollector
+	baseCollectorOnce sync.Once // Ensures that the baseCollector is initialized only once.
+	baseCollector     *BaseCollector
+
+	evictionCollectorOnce sync.Once
+	evictionCollector     *EvictionMetrics
 )
 
-// GetMetricsCollector returns the singleton MetricsCollector instance.
-// It panics if InitializeMetrics has not been called to ensure metrics are properly initialized.
-// Must be called after InitializeMetrics to avoid runtime issues.
-// If you do it before, BAD THINGS WILL HAPPEN.
-func GetMetricsCollector() *MetricsCollector {
-	if collector == nil {
-		panic("MetricsCollector not initialized. Call InitializeMetrics first.")
+// GetBaseMetricsCollector returns the singleton BaseCollector instance.
+func GetBaseMetricsCollector() *BaseCollector {
+	if baseCollector == nil {
+		panic("BaseCollector not initialized. Call InitializeMetrics first.")
 	}
-	return collector
+	return baseCollector
+}
+
+// GetEvictionMetricsCollector returns the singleton EvictionMetrics instance.
+func GetEvictionMetricsCollector() *EvictionMetrics {
+	if evictionCollector == nil {
+		panic("EvictionMetrics not initialized. Call InitializeMetrics first.")
+	}
+	return evictionCollector
```

Comment:
What's the advantage of having `evictionCollector` be private if there's a public getter function for it?